### ISSUE TITLE
Total execution time is not equal to the sum of individual times (komppile -v, krun -v)

### ISSUE THREAD

**Issue by denis-bogdanas** [kframework/k-legacy#169]
For benchmarking consideration it would be nice to have correct individual times for each component of K logic. And the sum of components should be equal to the total. 

Also, there are other issues with information displayed by -v option. example:

krun -v tests\diverse\collatz.simple

Two last log lines are displayed after input is read from console.
Platform: Windows.

**Comment by osa1** [comment]
> Two last log lines are displayed after input is read from console.

this part may be realted with flushing streams http://docs.oracle.com/javase/7/docs/api/java/io/OutputStream.html#flush()

**Comment by denis-bogdanas** [comment]
No, it is because execution time is incorrectly calculated. Will commit the fix right away.

**Comment by denis-bogdanas** [comment]
P.S. The second problem is not a problem after all. The order in which things are displayed in the output is correct. Just execution times were wrong.

**Comment by osa1** [comment]
@denis-bogdanas are you sure making stopwatch singleton is a good idea? It's used in lots of different places with different purposes so this change may break other things ...

**Comment by osa1** [comment]
@denis-bogdanas like I said, your change broke the build(see jenkins), we need to find another solution. I think what you need is to share stopwatch reference instead of making it singletion.

**Comment by denis-bogdanas** [comment]
Sorry for late answer. Was busy fixing the build yesterday.
1. Why stopwatch was made singleton? Because the purpose of this class should be to measure the time spent by different slices of logic of the running K process. K process is only one, time system is only one, so stopwatch should also be only one. I identified cases when stopwatch is used just to get the time difference between two execution moments. Well such usage seems antipattern to me and I replaced it with direct calls to System.currentTimeMillis(). In the end, the change I made is not a significant change in the architecture, even if I had to modify a lot of files. The need to modify a lot of files was a bigger problem - I guess it forced others to merge their changes with mine.
2. The build was broken because the execution times I added were printed always, not just when -v option was provided. So this was easy to fix.
